index.js

```diff
@@ -0,0 +1,223 @@
+const constants = {
+    'PI': 3.1415,
+    'e' : 2.7182
+}
+const variable = {}
+var variableExists = false
+var varEntry = null
+var elements = []
+var operators = []
+var mOperator = []
+var mElments = []
+
+function values(el){
+    let x = document.getElementById("answer");
+    var old = x.value;
+    if (!parseInt(el.value) && parseInt(el.value) !== 0){
+        x.value = old + " " + el.value + " ";
+    }
+    else{
+        x.value = old + el.value
+    }
+}
+
+function calculate(){
+    var x = document.getElementById("answer");
+    var array = x.value.split(" ")
+    array = array.filter(n => n)
```

Comment:
As the algorithm works like: if user give any operator input, it will append a space before and after the operator.
Means, 23 <space> + <space> 32

But, If user add single valued operator, it will append two spaces between them will be increase the length
Means, 23 <space> <space> sqrt <space> 25

Thats why, I had to filter all the valid values from array